beetmoverscript/script.py

```diff
@@ -86,6 +103,20 @@
             enrich_balrog_manifest(context.release_props, source, locale, destinations)
         )
 
+    context.checksums_manifest.extend(enrich_checksums_manifest(source, pretty_name))
+
+
+def enrich_checksums_manifest(path, pretty_name):
+    digest_info = []
+
+    size = get_size(path)
+    for algo in ("sha512", "md5", "sha1"):
```

Comment:
We may be calculating the same hash multiple times (`get_hash` here and line 115).
If we want to speed things up, we may want to create a dictionary with the various name, size, and hash info, and then populate templates with that information.